src/main/java/com/passbook/sparkeighteen/service/consumer/TransactionConsumer.java

```diff
@@ -0,0 +1,23 @@
+package com.passbook.sparkeighteen.service.consumer;
+
+import com.passbook.sparkeighteen.service.TransactionService;
+import com.passbook.sparkeighteen.utils.Constants;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+@Service
+public class TransactionConsumer {
+
+    private final TransactionService transactionService;
+
+    public TransactionConsumer(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    @KafkaListener(topics = Constants.TRANSACTION_REQUESTS_TOPIC, groupId = Constants.GROUP_ID)
+    public void consume(String txnRequest) throws IOException {
```

Comment:
Did you explore can you read from start from or from where you left or a random point?